### PR TITLE
LPS-157346 Add Date field as filter to Aggregation field

### PR DESCRIPTION
**Pull request by danvb112**
Related Issues:
[LPS-157346 - Adapt Filter Modal to Support Date with range operator](https://issues.liferay.com/browse/LPS-157346)
[LPS-157532 - Add Validations in Modal to add filter](https://issues.liferay.com/browse/LPS-157532)

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/EditObjectField.tsx [hunk 1]
```diff
@@ -226,23 +226,52 @@ export default function EditObjectField({
...
-							filterType,
-							json: {
-								[filterType as string]: value
-									? value
-									: valueList?.map(({value}) => value),
+			let newObjectFieldSettings: ObjectFieldSetting[] | undefined = [];
+
+			if (objectFieldBusinessType === 'Date') {
+				const dateJson: ObjectFieldDateRangeFilterSettring = {};
+
```
Comment (alinedoleron) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/EditObjectField.tsx:233]: I think there's a typo here on `ObjectFieldDateRangeFilterSettring` :grin:

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx [hunk 2]
```diff
@@ -352,6 +341,44 @@ export function ModalAddFilter({
...
 				)}
+
+				{selectedFilterBy?.businessType === 'Date' && (
+					<>
+						<Input
+							label={Liferay.Language.get('start')}
+							onChange={({target: {value}}) => {
+								setItems([
+									...items.filter(
+										(item) => item.value !== 'ge'
```
Comment (alinedoleron) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx:352]: Hi @danvb112. Does these `'ge'` and `'le'` terms change according to the language? Also, I think it's a little bit confusing when reading. Is there a way to make this more clear, like creating a variable or something else?

**Review by alinedoleron — CHANGES_REQUESTED** [review]
Hello @danvb112. Good job!
There are just two comment to you to check. Thanks! :relaxed:

**Comment by danvb112** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx (hunk 2 above)
Comment (danvb112) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx:418]: Hey @alinedoleron, these `ge` and `le` are related to the filter operator we need to send to the backend, we are following this Liferay documentation https://help.liferay.com/hc/en-us/articles/360031163631-Filter-Sort-and-Search

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx [hunk 3]
```diff
@@ -297,10 +339,15 @@ export function ModalAddFilter({
...
 							'there-are-no-columns-available'
 						)}
+						error={errors.selectedFilterBy}
 						items={filteredAvailableFields}
 						label={Liferay.Language.get('filter-by')}
 						onChangeQuery={setQuery}
-						onSelectItem={setSelectedFilterBy}
+						onSelectItem={(item) => {
+							setSelectedFilterBy(item);
+							setSelectedFilterType(undefined);
```
Comment (alinedoleron) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx:348]: Hi @danvb112. Once Milton reviewed my code and said that there's a difference between value null and undefined. I think in this case you should set to null, according to this definition (https://www.tektutorialshub.com/typescript/null-undefined-in-typescript/#:~:text=TypeOf-,Undefined%20%26%20Null,does%20not%20have%20a%20value.).
But please, feel free to make the change or not. You can ask Milton in the future hehe

> The value undefined means value is not assigned & you don’t know its value. It is an unintentional absence of value. It means that a variable has been declared but has not yet been assigned a value.

> The value null indicates that you know that the field does not have a value. It is an intentional absence of value.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx (hunk 2 above)
Comment (alinedoleron) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx:416]: Got it!! Thank you very much for the explanation!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx (hunk 3 above)
Comment (danvb112) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx]: Thanks @alinedoleron

**Comment by danvb112** [comment]
ci:stop

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Review by alinedoleron — APPROVED** [review]
LGTM!

**Comment by danvb112** [comment]
ci:forward

**Pull request closed without merging**